src/factories/schema-registry-factory.js

```diff
@@ -619,11 +628,13 @@ var SchemaRegistryFactory = function ($rootScope, $http, $location, $q, $log, Ut
       for (var i = 0; i < sortedHistory.length; i++) {
         var previous = '';
         if (i > 0)
-          previous = JSON.parse(sortedHistory[i - 1].schema);
+          previous = schemaParser(sortedHistory[i - 1].schema)
+
+        var current = schemaParser(sortedHistory[i].schema)
```

Comment:
We don't need to define this variable. Just change the line `current: JSON.parse(sortedHistory[i].schema)` to use our safe parser.